dts/arm/ti/mspm0g1xxx.dtsi

```diff
@@ -109,6 +109,19 @@
                     mode = <PERIODIC_DOWN>;
                 };
 
+		tima0: tima0@40860000 {
+                    compatible      = "ti,mspm0-pwm";
+                    status          = "okay";
+                    pinctrl-0          = <&adc1_pb17>;
+                    pinctrl-names      = "default";
+                    ti,clock-frequency = <32000000>;
+                    reg             = <0x40860000 0x2000>;
+                    ti,divide-ratio    = <RATE_1>;
+                    ti,prescaler       = <0>;
+                    ti,mode            = <PWM_EDGE_ALIGN_UP>;
+                    ti,cc_index        = <2>;
+		};
```

Comment:
```suggestion
		tima0: tima0@40860000 {
                    compatible      = "ti,mspm0-pwm";
                    status          = "disabled";
                    ti,clock-frequency = <32000000>;
                    reg             = <0x40860000 0x2000>;
                    ti,divide-ratio    = <RATE_1>;
                    ti,prescaler       = <0>;
                    ti,mode            = <PWM_EDGE_ALIGN_UP>;
		};
```
1. the timer shouldn't be enabled by the base devicetree.

2. the cc_index shouldn't be set by the base devicetree - how does it know anything about what channel you want?
    Also, I think the cc stuff actually belongs to the consumers of pwm devices exposed by this driver, i.e. your puc stuff would take a phandle `zephyr,pwm = <&tima0 2>` to grab cc_index 2 or whatever.
    
3. the pinctrl shouldn't be set by the base devicetree - how does it know anything about how you want the IOs configured?